### PR TITLE
Add and Sub command support 

### PR DESCRIPTION
**Pull request by alex8937**
image

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -482,6 +483,66 @@ func (c *raftKVClient) OptimizeTxnCommands() {
 	c.txnCmds.Commands = newCmds
 }
 
+func (c *raftKVClient) AddTransaction(cmdArr []string ) error {
+	amount, _ := parseInt64(cmdArr[2])
```
Comment (renga86) [inline client/client.go:487]: nit: is `err` already being handled upstream ?

File: client/client.go [hunk 2]
```diff
@@ -482,6 +483,66 @@ func (c *raftKVClient) OptimizeTxnCommands() {
...
+func (c *raftKVClient) attemptAdd(key string, amount int64) error {
+	c.txnCmds = &raftpb.RaftCommand{
+		Commands: []*raftpb.Command{
+			{Method: common.GET, Key: key},
+		},
+		IsTxn: true,
+	}
+
+	// Send txn to the server & fetch the response
+	getTxnRsp, err := c.Transaction()
```
Comment (renga86) [inline client/client.go:520]: qq: why can't this be simply `GET` ? Why do we need to call Transaction and then disable the `txn` flag ?

File: client/client.go (hunk 1 above)
Comment (alex8937) [reply client/client.go:487]: Yeah. Err is handled in 
```
	case common.SET, common.ADD, common.SUB:
		return c.validCmd3(cmdArr)
```

File: client/client.go (hunk 2 above)
Comment (alex8937) [reply client/client.go:520]: B/c simple `GET` does not provide you with `getTxnRsp`. Instead, it returns you a string like `Key=x, Value=100`.

File: client/client.go (hunk 1 above)
Comment (alex8937) [reply client/client.go:487]: https://github.com/SCPD-Project/raft-kv-store/blob/f53538048a1fe834e679be334ebb49cb32db267f/client/client.go#L161

File: client/client.go (hunk 2 above)
Comment (renga86) [reply client/client.go:520]: hmm sending it as "non transaction" will make less state to maintain at server. Anyhow approving in the interest of time.

**Pull request merged**